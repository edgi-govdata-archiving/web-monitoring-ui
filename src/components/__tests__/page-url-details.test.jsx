/* eslint-env jest */
import { render } from '@testing-library/react';
import PageUrlDetails from '../page-url-details/page-url-details';
import simplePage from '../../__mocks__/simple-page.json';

describe('PageUrlDetails Component', () => {
  // Change string values to date objects so they're parsed correctly
  simplePage.versions.forEach(version => {
    version.capture_time = new Date(version.capture_time);
  });

  it('shows nothing if version URLs and redirects match the page URL', () => {
    const { container } = render(
      <PageUrlDetails
        page={simplePage}
        from={simplePage.versions[1]}
        to={simplePage.versions[0]}
      />
    );

    expect(container).toBeEmptyDOMElement();
  });

  it("shows the versions' URL if it differs from the page's", () => {
    const version1 = {
      ...simplePage.versions[1],
      url: `${simplePage.url}/something`
    };
    const version2 = {
      ...simplePage.versions[0],
      url: `${simplePage.url}/something`
    };
    const { container } = render(
      <PageUrlDetails
        page={simplePage}
        from={version1}
        to={version2}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it("shows the versions' redirects", () => {
    const version1 = {
      ...simplePage.versions[1],
      source_metadata: {
        ...simplePage.versions[1].source_metadata,
        redirects: [
          simplePage.url,
          `${simplePage.url}/something`
        ]
      }
    };
    const version2 = {
      ...simplePage.versions[0],
      source_metadata: {
        ...simplePage.versions[0].source_metadata,
        redirects: [
          simplePage.url,
          `${simplePage.url}/something`
        ]
      }
    };
    const { container } = render(
      <PageUrlDetails
        page={simplePage}
        from={version1}
        to={version2}
      />
    );

    expect(container).toMatchSnapshot();
  });

  it('shows separate URL histories for each version if they differ', () => {
    const version1 = {
      ...simplePage.versions[1],
      source_metadata: {
        ...simplePage.versions[1].source_metadata,
        redirects: [
          simplePage.url,
          `${simplePage.url}/something`
        ]
      }
    };
    const version2 = {
      ...simplePage.versions[0],
      source_metadata: {
        ...simplePage.versions[0].source_metadata,
        redirects: [
          simplePage.url,
          `${simplePage.url}/something/else`
        ]
      }
    };
    const { container } = render(
      <PageUrlDetails
        page={simplePage}
        from={version1}
        to={version2}
      />
    );

    expect(container).toMatchSnapshot();
  });
});
