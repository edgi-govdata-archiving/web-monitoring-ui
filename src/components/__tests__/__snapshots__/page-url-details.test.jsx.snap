// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`PageUrlDetails Component shows separate URL histories for each version if they differ 1`] = `
<div>
  <details
    class="pageUrlDetails"
  >
    <summary>
      ⚠️ Versions come from different URLs
    </summary>
    <strong>
      From Version URL and Redirects:
    </strong>
    <ol
      class="urlHistoryList"
    >
      <li>
        <a
          href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://
           
          www
          .
          ncei
          .
          noaa
          .
          gov
           
          / news
           
          / earth-science-conference-convenes
           
          
           
          
        </a>
        <i
          class="fa fa-no-hover fa-right-icon fa-angle-right"
          title="Redirects to"
        />
      </li>
      <li>
        <a
          href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes/something"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://
           
          www
          .
          ncei
          .
          noaa
          .
          gov
           
          / news
           
          / earth-science-conference-convenes
           
          <ins>
            / something
          </ins>
           
          
           
          
        </a>
      </li>
    </ol>
    <strong>
      To Version URL and Redirects:
    </strong>
    <ol
      class="urlHistoryList"
    >
      <li>
        <a
          href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://
           
          www
          .
          ncei
          .
          noaa
          .
          gov
           
          / news
           
          / earth-science-conference-convenes
           
          
           
          
        </a>
        <i
          class="fa fa-no-hover fa-right-icon fa-angle-right"
          title="Redirects to"
        />
      </li>
      <li>
        <a
          href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes/something/else"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://
           
          www
          .
          ncei
          .
          noaa
          .
          gov
           
          / news
           
          / earth-science-conference-convenes
           
          <ins>
            / something
          </ins>
           
          <ins>
            / else
          </ins>
           
          
           
          
        </a>
      </li>
    </ol>
  </details>
</div>
`;

exports[`PageUrlDetails Component shows the versions' URL if it differs from the page's 1`] = `
<div>
  <details
    class="pageUrlDetails"
  >
    <summary>
      ⚠️ Captured from a different URL
    </summary>
    <a
      href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes/something"
      rel="noopener noreferrer"
      target="_blank"
    >
      http://
       
      www
      .
      ncei
      .
      noaa
      .
      gov
       
      / news
       
      / earth-science-conference-convenes
       
      <ins>
        / something
      </ins>
       
      
       
      
    </a>
  </details>
</div>
`;

exports[`PageUrlDetails Component shows the versions' redirects 1`] = `
<div>
  <details
    class="pageUrlDetails"
  >
    <summary>
      ⚠️ Captured URL Redirected Somewhere Else
    </summary>
    <ol
      class="urlHistoryList"
    >
      <li>
        <a
          href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://
           
          www
          .
          ncei
          .
          noaa
          .
          gov
           
          / news
           
          / earth-science-conference-convenes
           
          
           
          
        </a>
        <i
          class="fa fa-no-hover fa-right-icon fa-angle-right"
          title="Redirects to"
        />
      </li>
      <li>
        <a
          href="http://www.ncei.noaa.gov/news/earth-science-conference-convenes/something"
          rel="noopener noreferrer"
          target="_blank"
        >
          http://
           
          www
          .
          ncei
          .
          noaa
          .
          gov
           
          / news
           
          / earth-science-conference-convenes
           
          <ins>
            / something
          </ins>
           
          
           
          
        </a>
      </li>
    </ol>
  </details>
</div>
`;
